src/ui/public/vis/editors/default/keyboard_move.js

```diff
@@ -0,0 +1,57 @@
+/**
+ * The keyboardMove directive can be attached to elements, that can receive keydown events.
+ * It will call the passed callback function and pass the direction in which an
+ * arrow key was pressed to the callback (as the argument with the name `direction`).
+ * The passed value will be one of `Direction.up` or `Direction.down`, which can be
+ * imported to compare against those values. The directive will also make sure, that
+ * the pressed button will get the focus back (e.g. if it was lost due to a ng-repeat
+ * reordering).
+ *
+ * Usage example:
+ *
+ * <button keyboard-move="onMoved(direction)">...</button>
+ *
+ * import { Direction } from './keyboard_move';
+ * function onMoved(dir) {
+ *   if (dir === Direction.up) {
+ *     // moved up
+ *   } else if (dir === Direction.down) {
+ *     // moved down
+ *   }
+ * }
+ */
+import { uiModules } from 'ui/modules';
+
+export const Direction = {
+  up: 'up',
+  down: 'down'
+};
+
+const directionMapping = {
+  38: Direction.up,
+  40: Direction.down
+};
+
+uiModules.get('kibana')
+.directive('keyboardMove', ($timeout) => ({
```

Comment:
Since this directive really adds some logic to an element, and doesn't just listen on an event, that would anyway happen, I would rather not rename it `onKeyboardMove`. A directive named `on-` sounds for me side effect free, which this isn't. Besides Angular itself not naming any of the listeners `on-`, but just `ng-keydown`, etc.

Would you be up for discussion on this topic?